### PR TITLE
Upgrade client go to kubernetes-1.12.4

### PR DESCRIPTION
**Pull request by fisherxu**
See [comments here](https://github.com/kubernetes-sigs/federation-v2/pull/394#issuecomment-449002333) .

/hold

### CODE REVIEW

**Comment by shashidharatd** [comment]
This also partially addresses https://github.com/kubernetes-sigs/federation-v2/issues/83 and still uses dynamic client. This LGTM as an intermediate step before migrating new kubebuilder 1.0.6. Would request @font / @marun to approve.

/lgtm

File: test/e2e/crd.go [hunk 1]
```diff
@@ -23,21 +23,20 @@ import (
 	"github.com/pborman/uuid"
 
 	apicommon "github.com/kubernetes-sigs/federation-v2/pkg/apis/core/common"
+	fedv1a1 "github.com/kubernetes-sigs/federation-v2/pkg/apis/core/v1alpha1"
```
Comment (marun) [inline test/e2e/crd.go]: It is an (unfortunately) undocumented goal to format imports as follows:

```
[system imports]

[3rd party imports (e.g. uuid, glog)]

[kube and community imports]

[federation imports]

[testing imports]
```

File: test/e2e/crd.go [hunk 2]
```diff
@@ -103,14 +102,13 @@ func validateCrdCrud(f framework.FederationFramework, targetCrdKind string, name
 	userAgent := fmt.Sprintf("test-%s-crud", strings.ToLower(targetCrdKind))
 
 	// Create the target crd in all clusters
-	var pools []dynamic.ClientPool
-	var hostPool dynamic.ClientPool
+	var configs []*rest.Config
+	var config *rest.Config
```
Comment (marun) [inline test/e2e/crd.go]: s/config/hostConfig/ for clarity.

**Comment by marun** [comment]
Thank you for doing this!  I have a minor nit about import ordering inline, but otherwise lgtm.

File: Gopkg.toml [hunk 3]
```diff
@@ -24,11 +24,24 @@ required = ["github.com/kubernetes/repo-infra/verify/boilerplate/test"]
...
   name="k8s.io/apimachinery"
-  version="kubernetes-1.10.1"
+  version="kubernetes-1.12.4"
 
 [[override]]
   name="k8s.io/client-go"
-  version="kubernetes-1.10.1"
+  version="kubernetes-1.12.4"
+
+[[override]]
```
Comment (gyliu513) [inline Gopkg.toml]: A question here: Why did you add another three overrides here?

Comment (fisherxu) [reply Gopkg.toml]: If we just upgrade the client-go, this three older repo will not work, they should keep the same version :)

Comment (gyliu513) [reply Gopkg.toml]: Got it, thanks @fisherxu :-)

File: test/e2e/crd.go (hunk 2 above)
Comment (marun) [reply test/e2e/crd.go]: Oh, that's because they are the same value.  Please remove the `config` variable and ensure instances of `hostPool` are changed to `hostConfig`.

Comment (fisherxu) [reply test/e2e/crd.go]: have done :-)

**Comment by marun** [comment]
When I try to run `dep ensure` locally against this PR, I see the following error:

```
grouped write of manifest, lock and vendor: failed to export k8s.io/client-go: fatal: failed to unpack tree object 5e6a3d4e34f694e895b13ae728111e726a5b69df
: exit status 128
```

I don't have this problem against master, which suggests that this PR may be leaving the tree in a problematic state for future vendoring.  I'm putting a hold on the PR until I can rule out that possibility.

/hold

**Comment by marun** [comment]
False alarm.

/hold cancel
/approve
/lgtm
